include/dba.php

```diff
@@ -1016,16 +1016,12 @@ public static function delete($table, $param, $in_process = false, &$callstack =
 	public static function update($table, $fields, $condition, $old_fields = array()) {
 		$table = self::escape($table);
 
-		if (count($condition) > 0) {
-			$array_element = each($condition);
-			$array_key = $array_element['key'];
-			if (is_int($array_key)) {
-				$condition_string = " WHERE ".array_shift($condition);
-			} else {
-				$condition_string = " WHERE `".implode("` = ? AND `", array_keys($condition))."` = ?";
-			}
+		$array_element = each($condition);
```

Comment:
What if $condition is an empty array?

Reply:
When the condition is empty, the system will generate a warning and the generated SQL will be invalid. This is a safety thing. I already had destroyed my contact table in the past with an update query with an empty condition before this. Especially when now masses of queries will be changed, it can always happen that there will be bugs like these.

Reply:
Understood, however I don't like this idea of relying on PHP warnings and invalid SQL strings for input validation. I'd rather we use a actual user-triggered fatal error or an Exception for that.